mllib/src/main/scala/org/apache/spark/ml/stat/Summarizer.scala

```diff
@@ -94,46 +97,86 @@ object Summarizer extends Logging {
    *  - min: the minimum for each coefficient.
    *  - normL2: the Euclidian norm for each coefficient.
    *  - normL1: the L1 norm of each coefficient (sum of the absolute values).
-   * @param firstMetric the metric being provided
-   * @param metrics additional metrics that can be provided.
+   * @param metrics metrics that can be provided.
    * @return a builder.
    * @throws IllegalArgumentException if one of the metric names is not understood.
    *
    * Note: Currently, the performance of this interface is about 2x~3x slower then using the RDD
    * interface.
    */
   @Since("2.3.0")
-  def metrics(firstMetric: String, metrics: String*): SummaryBuilder = {
-    val (typedMetrics, computeMetrics) = getRelevantMetrics(Seq(firstMetric) ++ metrics)
+  def metrics(metrics: String*): SummaryBuilder = {
```

Comment:
Use `def metrics(metrics: String*)` instead of `def metrics(firstMetric: String, metrics: String*)`.
It will make pyspark call this interface more easier. (Later I will add python API)

Reply:
have you tried about java? IIRC this style is for java compatibility.

Reply:
+1 @cloud-fan

Reply:
I haven't test this on Java. But, I can find some other places use similar style, such as `Dataset.toDF`, `Dataset.drop`, Does it mean they also have java compatibility issue ?

Reply:
@cloud-fan Do you say about this bug ? https://issues.apache.org/jira/browse/SPARK-5904
But it is only related to abstract method.
Now I add java testsuite to make sure it works fine.